tools/ctl/ide/ui.go

```diff
@@ -159,26 +172,67 @@ func diffArtifacts(ctx context.Context, example rundex.Rebuild) {
 		log.Println(errors.Wrap(err, "failed to create debug asset store"))
 		return
 	}
-	// TODO: Clean up these artifacts.
-	// TODO: Check if these are already downloaded.
-	rebuildAsset := rebuild.DebugRebuildAsset.For(t)
-	upstreamAsset := rebuild.DebugUpstreamAsset.For(t)
-	rba := localAssets.URL(rebuildAsset).Path
-	usa := localAssets.URL(upstreamAsset).Path
-	if _, err := os.Stat(rba); errors.Is(err, os.ErrNotExist) {
-		if err := rebuild.AssetCopy(ctx, localAssets, debugAssets, rebuildAsset); err != nil {
-			log.Println(errors.Wrap(err, "failed to copy rebuild asset"))
+	var rba, usa string
+	wasSmoketest := example.ObliviousID == ""
+	if wasSmoketest {
+		// TODO: Clean up these artifacts.
+		rebuildAsset := rebuild.DebugRebuildAsset.For(t)
+		upstreamAsset := rebuild.DebugUpstreamAsset.For(t)
+		rba = localAssets.URL(rebuildAsset).Path
+		usa = localAssets.URL(upstreamAsset).Path
+		if _, err := os.Stat(rba); errors.Is(err, os.ErrNotExist) {
+			if err := rebuild.AssetCopy(ctx, localAssets, debugAssets, rebuildAsset); err != nil {
+				log.Println(errors.Wrap(err, "failed to copy rebuild asset"))
+				return
+			}
+			log.Printf("downloaded rebuild: %s", rba)
+		}
+		if _, err := os.Stat(usa); errors.Is(err, os.ErrNotExist) {
+			if err := rebuild.AssetCopy(ctx, localAssets, debugAssets, upstreamAsset); err != nil {
+				log.Println(errors.Wrap(err, "failed to copy upstream asset"))
+				return
+			}
+			log.Printf("downloaded upstream: %s", usa)
+		}
+	} else {
+		r, err := debugAssets.Reader(ctx, rebuild.Asset{Target: t, Type: rebuild.BuildInfoAsset})
+		if err != nil {
+			log.Println(errors.Wrap(err, "reading build info"))
 			return
 		}
-	}
-	if _, err := os.Stat(usa); errors.Is(err, os.ErrNotExist) {
-		if err := rebuild.AssetCopy(ctx, localAssets, debugAssets, upstreamAsset); err != nil {
-			log.Println(errors.Wrap(err, "failed to copy upstream asset"))
+		bi := rebuild.BuildInfo{}
+		if err := json.NewDecoder(r).Decode(&bi); err != nil {
+			log.Println(errors.Wrap(err, "decoding build info"))
 			return
 		}
-		log.Printf("downloaded rebuild and upstream:\n\t%s\n\t%s", rba, usa)
+		metadata, _ := rebuild.NewGCSStore(context.WithValue(ctx, rebuild.RunID, bi.ID), "gs://rebuild-metadata")
+		// TODO: Clean up these artifacts.
+		rebuildAsset := rebuild.RebuildAsset.For(t)
+		upstreamAsset := rebuild.DebugUpstreamAsset.For(t)
+		rba = localAssets.URL(rebuildAsset).Path
+		usa = localAssets.URL(upstreamAsset).Path
+		if _, err := os.Stat(rba); errors.Is(err, os.ErrNotExist) {
```

Comment:
this breaks the abstraction layer with localAssets, right? would it be possible to try to read from the localAssets and, on failure, do a copy?

Reply:
Done.